Lib/multiprocessing/shared_memory.py

```diff
@@ -95,6 +96,7 @@ def __init__(self, name=None, create=False, size=0):
                     self._name = name
                     break
             else:
+                name = "/" + name if self._prepend_leading_slash else name
```

Comment:
One thing to consider. This will make it impossible to attach to a "foo" literal (e.g. a memory name which was not created by this class). The `_prepend_leading_slash` class attribute can be toggled to allow this, therefore I think it should be a public attribute (not worth being documented).

Reply:
Actually I'm not sure if this use-case should be supported (and it looks pretty rare anyway). Having this as a class settable attribute is good already IMO. And if there's demand we can introduce it as a public property later. Dismiss my suggestion.

Reply:
Good point.  I was worried about needing to clarify that it only impacted POSIX systems and had no effect on Windows.  At the same time, I saw value exposing it as a hook for users to control in a subclass (given that the leading slash is really only needed on FreeBSD/NetBSD and otherwise shortens the remaining space for the name).  I thought to be cautious and start by making it private but maybe it should be made public?

Reply:
IMO it should remain private as per your point about Windows. If there ever will be demand for it we can simply add it as a `prepend_leading_slash` public property (with setter) so that we won't break code for those few people who in the meantime relied on `_prepend_leading_slash`.

Reply:
Cool -- sounds good.